### ISSUE TITLE
Unable to open input files

### ISSUE THREAD

**Issue by tobias-tobias** [performous/performous#169]
Hallo everybody,
I installed perfomous from source using Opensuse 13.2. After starting the executable no song formats could be open, even those and also after make install:

ffmpeg/error: Failed to open "/home/tobias/bin/performous/performous/data/sounds/drum_bass.ogg": Cannot open input file

I found out, that it is the avformat_open_input() function in ffmpeg.cc that causes the error. I also found out that it is the AVERROR -1094995529, which means "Invalid data found when processing input".

Doing make causes this output, without interrupting:

In file included from /home/tobias/bin/performous/performous/game/ffmpeg.cc:12:0:
/usr/include/ffmpeg/libavcodec/avcodec.h: In constructor ‘AVPacket::AVPacket()’:
/usr/include/ffmpeg/libavcodec/avcodec.h:1400:16: warning: ‘AVPacket::destruct’ is deprecated (declared at /usr/include/ffmpeg/libavcodec/avcodec.h:1444) [-Wdeprecated-declarations]
 typedef struct AVPacket {
                ^
/usr/include/ffmpeg/libavcodec/avcodec.h:1400:16: warning: ‘AVPacket::priv’ is deprecated (declared at /usr/include/ffmpeg/libavcodec/avcodec.h:1446) [-Wdeprecated-declarations]
/home/tobias/bin/performous/performous/game/ffmpeg.cc: In constructor ‘FFmpeg::decodePacket()::ReadFramePacket::ReadFramePacket(AVFormatContext*)’:
/home/tobias/bin/performous/performous/game/ffmpeg.cc:181:45: note: synthesized method ‘AVPacket::AVPacket()’ first required here 
   ReadFramePacket(AVFormatContext* s): m_s(s) {
                                             ^

I hope someone can find the error.

Kind regards,
Tobias

**Comment by TobiX** [comment]
Looks a bit like a broken ffmpeg installation to me...

**Comment by tobias-tobias** [comment]
So what can I do? How can I test it?

**Comment by tobias-tobias** [comment]
The libav* libraries are all version 56, but only libavutil is 54
could this be a reason? I didn't find newer one.

**Comment by TobiX** [comment]
Now, those numbers are matching versions, see for example the list of binaries in Debian: https://tracker.debian.org/pkg/ffmpeg

**Comment by nieknooijens** [comment]
In file included from /home/tobias/bin/performous/performous/game/ffmpeg.cc:12:0:
/usr/include/ffmpeg/libavcodec/avcodec.h: In constructor ‘AVPacket::AVPacket()’:

it's not from ffmpeg.cc in performous, it's from "a file included" by us, so externally.
don't know if the problem occurs because ubuntu runs libAV and suse runs ffmpeg, but ffmpeg should be 100% compatible since it also comes with libavresample!

**Comment by tobias-tobias** [comment]
Yes, but the error is:

ffmpeg/error: Failed to open "/home/tobias/bin/performous/performous/data/sounds/drum_bass.ogg": Cannot open input file

and this is from ffmpeg.cc

**Comment by tobias-tobias** [comment]
there some other things:
I have to start the jack server manually to use my sound card
When started, I can't hear other music parallel

when jack stopped again my Kaffein program strikes with switching TV channels

... I tried a lot to find out, why I can't open music files.

**Comment by nieknooijens** [comment]
oh but performous binds to pulseaudio so it shouldn't have any connection with JACK

**Comment by tobias-tobias** [comment]
I think the Jack error is independent of the open input file error

the output without jack started:

logger/notice: Logging all notices, warnings and errors. Log file: /home/tobias/.cache/performous/infolog.txt
core/notice: Performous 1.0-108-gf379c90 starting...
  Internationalization: Enabled
  MIDI Hardware I/O:    Enabled
  Webcam support:       Enabled
core/notice: Starting the audio subsystem (errors printed on console may be ignored).
ALSA lib pcm.c:2259:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.rear
ALSA lib pcm.c:2259:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.center_lfe
ALSA lib pcm.c:2259:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.side
ALSA lib pcm_route.c:867:(find_matching_chmap) Found no matching channel map
ALSA lib pcm_route.c:867:(find_matching_chmap) Found no matching channel map
Cannot connect to server socket err = No such file or directory
Cannot connect to server request channel
jack server is not running or cannot be started
audio/error: Audio device 'dev="system" mics=blue': No such device.
audio/error: Audio device 'dev="system" out=2': No such device.
ffmpeg/error: Failed to open "/usr/local/share/games/performous/sounds/drum_bass.ogg": Cannot open input file
ffmpeg/error: Failed to open "/usr/local/share/games/performous/sounds/drum_hi-hat.ogg": Cannot open input file
ffmpeg/error: Failed to open "/usr/local/share/games/performous/sounds/drum_tom1.ogg": Cannot open input file
ffmpeg/error: Failed to open "/usr/local/share/games/performous/sounds/drum_cymbal.ogg": Cannot open input file
ffmpeg/error: Failed to open "/usr/local/share/games/performous/sounds/guitar_fail1.ogg": Cannot open input file
ffmpeg/error: Failed to open "/usr/local/share/games/performous/sounds/guitar_fail2.ogg": Cannot open input file
ffmpeg/error: Failed to open "/usr/local/share/games/performous/sounds/drum_snare.ogg": Cannot open input file
ffmpeg/error: Failed to open "/usr/local/share/games/performous/sounds/guitar_fail3.ogg": Cannot open input file
ffmpeg/error: Failed to open "/usr/local/share/games/performous/sounds/guitar_fail4.ogg": Cannot open input file
ffmpeg/error: Failed to open "/usr/local/share/games/performous/sounds/guitar_fail5.ogg": Cannot open input file
ffmpeg/error: Failed to open "/usr/local/share/games/performous/sounds/guitar_fail6.ogg": Cannot open input file
ffmpeg/error: Failed to open "/usr/local/share/games/performous/themes/default/menu.ogg": Cannot open input file


and now with started jack server (jack_control start):

logger/notice: Logging all notices, warnings and errors. Log file: /home/tobias/.cache/performous/infolog.txt
core/notice: Performous 1.0-108-gf379c90 starting...
  Internationalization: Enabled
  MIDI Hardware I/O:    Enabled
  Webcam support:       Enabled
core/notice: Starting the audio subsystem (errors printed on console may be ignored).
ALSA lib pcm_dsnoop.c:618:(snd_pcm_dsnoop_open) unable to open slave
ALSA lib pcm_dmix.c:1022:(snd_pcm_dmix_open) unable to open slave
ALSA lib pcm.c:2259:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.rear
ALSA lib pcm.c:2259:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.center_lfe
ALSA lib pcm.c:2259:(snd_pcm_open_noupdate) Unknown PCM cards.pcm.side
ALSA lib pcm_dmix.c:1022:(snd_pcm_dmix_open) unable to open slave
ffmpeg/error: Failed to open "/usr/local/share/games/performous/sounds/drum_bass.ogg": Cannot open input file
ffmpeg/error: Failed to open "/usr/local/share/games/performous/sounds/drum_snare.ogg": Cannot open input file
ffmpeg/error: Failed to open "/usr/local/share/games/performous/sounds/drum_hi-hat.ogg": Cannot open input file
ffmpeg/error: Failed to open "/usr/local/share/games/performous/sounds/drum_tom1.ogg": Cannot open input file
ffmpeg/error: Failed to open "/usr/local/share/games/performous/sounds/drum_cymbal.ogg": Cannot open input file
ffmpeg/error: Failed to open "/usr/local/share/games/performous/sounds/guitar_fail1.ogg": Cannot open input file
ffmpeg/error: Failed to open "/usr/local/share/games/performous/sounds/guitar_fail2.ogg": Cannot open input file
ffmpeg/error: Failed to open "/usr/local/share/games/performous/sounds/guitar_fail3.ogg": Cannot open input file
ffmpeg/error: Failed to open "/usr/local/share/games/performous/sounds/guitar_fail4.ogg": Cannot open input file
ffmpeg/error: Failed to open "/usr/local/share/games/performous/sounds/guitar_fail5.ogg": Cannot open input file
ffmpeg/error: Failed to open "/usr/local/share/games/performous/sounds/guitar_fail6.ogg": Cannot open input file
ffmpeg/error: Failed to open "/usr/local/share/games/performous/themes/default/menu.ogg": Cannot open input file

**Comment by nieknooijens** [comment]
so you don't have any ALSA/pulseaudio stack? since I see alsa errors not being able to open up the sound card.

**Comment by tobias-tobias** [comment]
... as I mentioned I read a lot of bug reports. So I tried a lot. I think that's making it complicated for you, sorry

**Comment by tobias-tobias** [comment]
how can I find out the thing with the ALSA/pulseaudio stack?

**Comment by nieknooijens** [comment]
type "alsamixer" into a terminal, it should look like this:
![schermafdruk van 2015-11-21 14-08-39](https://cloud.githubusercontent.com/assets/2794331/11318597/6d7bb108-9059-11e5-9b8d-3dc9895e835a.png)
this is my HDMI out so I press F6 to select my other card then it shows this:
![schermafdruk van 2015-11-21 14-09-47](https://cloud.githubusercontent.com/assets/2794331/11318602/8ad2adce-9059-11e5-97e5-9a2f56b5c561.png)
this usually means ALSA is running, you can use VLC/Totem to play a video and it should have audio WITHOUT jack running.

**Comment by nieknooijens** [comment]
just to be sure, that it isn't the audio stack itself that fails.

**Comment by tobias-tobias** [comment]
my standard card is called pulseaudio: there is only one balk.

And I can select HDA-Intel PCH with Chip Intel PantherPoint HDMI and there are more balks

Playing a video with VLC: there is sound, but no video image. In the KDE settings there is no device to chose for video recording in the audio and video settings

**Comment by tobias-tobias** [comment]
... kaffeine works with the same video

**Comment by tobias-tobias** [comment]
shall I do some pictures of the alsamixer output?

**Comment by nieknooijens** [comment]
well that really does look like a broken ffmpeg installation! VLC depends on ffmpeg for video/audio
http://packages.ubuntu.com/wily/vlc
perhaps installing the latest ffmpeg from source?

**Comment by tobias-tobias** [comment]
OK, there performous settings:

I started the program with
![bildschirmfoto9-klein](https://cloud.githubusercontent.com/assets/15947880/11318754/7a0768f8-905f-11e5-9efd-ce145645c28e.png)

**Comment by tobias-tobias** [comment]
and the files could not be open too with this configuration:
![bildschirmfoto10-klein](https://cloud.githubusercontent.com/assets/15947880/11318761/b4dd78dc-905f-11e5-8470-177802fb602e.png)

**Comment by nieknooijens** [comment]
yeah it really does look like broken ffmpeg.
this might help:
https://trac.ffmpeg.org/wiki/CompilationGuide
it's a pain but it'll probably fix it, recompiling performous will help too.

**Comment by tobias-tobias** [comment]
VLC worked for me until I made distro update on wednesday and this update was aborted at the very end.
Alsa is installed, but I didn't install the alsa-devel package. Do I need this?

**Comment by tobias-tobias** [comment]
There were always errors produced by having different repos for dependent packages, e.g. packman and opensuse repo
I will try to find out, if all dependencies are from the same repo ...

**Comment by tobias-tobias** [comment]
Hello everybody,

first of all there went something wrong with the distro-update - most of my packages were installed from additional repo called "home:/escos". So I repeated and changed all dependencies to "normal opensuse repos"

after this I checked VLC -  i'm using two GPUs. So there went something wrong with the "hardware acceleration". So VLC error does not depend on ffmpeg installation.

I compiled again and the file could not be open as well.


There are some differs in the sound architecture? ALSA is installed but not used? How can I change this? The performous says again Jack can't be started.

I hope the distro update didn't went wrong so much )-:

KInd regards

**Comment by TobiX** [comment]
On Linux, ALSA is always used. There is no other sound output. But you shouldn't need Jack. The "normal" Linux audio stack currently is App->Pulseaudio->ALSA->Hardware or App->ALSA->Hardware, for professional audio it is App->Jack->ALSA->Hardware, but you shouldn't need Jack for Performous.

And the opening of the input files is quite certainly not related to ALSA/Pulseaudio/Jack at all. I still think your system is broken in some horrible way, if it can't even play Ogg Vorbis via ffmpeg (do these files play with ffplay, mplayer, mplayer2 or mpv?)

**Comment by nieknooijens** [comment]
@TobiX yeah I think that's the case here, I just wanted to exclude the possibility that his audio stack was b0rked.

**Comment by tobias-tobias** [comment]
Hallo again,

yes the error was related to a corrupt OS Installation. I did the Opensuse update wrong!!!

After some tries I let check the hardware and after that I installed Opensuse13.2 new from DVD. Then I just installed the ffmpeg-devel package from packman without the libav*-devel packages. Now performous works fine (-:

Thank you for your help and thank you for the program at all! It's a really nice application. Now I just need to find the right music+txt files (-;

Kind regards,
Tobias
